### PR TITLE
treewide: remove {dclocal_,}read_repair_chance options

### LINKED ISSUE #3502: Remove probabilistic read repair
This is a mirror for [CASSANDRA-13910](https://issues.apache.org/jira/browse/CASSANDRA-13910).

While reads are typically sent to just enough replicas to satisfy the CL, read requests may be randomly sent to all replicas in order to repair potentially inconsistent data, based on ``read_repair_chance`` and ``dclocal_read_repair_chance`` (another exception is speculative retry).

Often this extra load is not justifiable, since inconsistencies are reliably repaired by either QUORUM reads or by hints. When there are dropped hints (something the manager should pay attention to, btw), then issuing a full repair is in order.

Other good arguments are given in the linked issue.

### PR DESCRIPTION
**Pull request by tchaikov**
dclocal_read_repair_chance and read_repair_chance have been removed in Cassandra 3.11 and 4.x, see
https://issues.apache.org/jira/browse/CASSANDRA-13910. if we expose these properties via DDL, Cassandra would fail to consume the CQL statement creating the table when performing migration from Scylla to Cassandra 4.x, as the latter does not understand these properties anymore.

currently the default values of `dc_local_read_repair_chance` and `read_repair_chance` are both "0". so they are practically disabled, unless user deliberately set them to a value greater than 0.

also, as a side effect, Cassandra 4.x has better support of Python3. the cqlsh shipped along with Cassandra 3.11.16 only supports python2.7, see
https://github.com/apache/cassandra/blob/cassandra-3.11.16/bin/cqlsh.py it errors out if the system only provides python3 with the error of
```
No appropriate python interpreter found.
```
but modern linux systems do not provide python2 anymore.

so, in this change, we deprecate these two options.

Fixes #3502
Signed-off-by: Kefu Chai <kefu.chai@scylladb.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
i will drop "dc_local_read_repair_chance" and "read_repair_chance" from tests first.

**Comment by avikivity** [comment]
What about documentation? And the storage_proxy parts?

**Comment by tchaikov** [comment]
v2: 

- drop / update related document
- remove `storage_proxy::new_read_repair_decision()`. it does not make sense to keep it, as it always returns `db::read_repair_decision::NONE` if these two options do not take effects.

> What about documentation? And the storage_proxy parts?

@avikivity should be addressed in the latest revision. could you take another look?

**Comment by tchaikov** [comment]
> This isn't deprecation. When something is deprecated, it still works, but it is recommended not to use it.
> 
> Here, you completely remove it.

agreed. i reworded the title of the pull request and the title of the first commit to reflect the intention of the change.

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7905/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7905/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/8902/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/872/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7905/)
> 
> #### Failed Tests (2/30246):
> * [auth_cluster.test_auth_v2_migration.debug.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/7905/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___auth_cluster_test_auth_v2_migration_debug_3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+auth_cluster.test_auth_v2_migration.debug.3)
> * [test_auth_v2_migration](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/7905/testReport/junit/%28root%29/test_auth_v2_migration/Tests___Unit_Tests___test_auth_v2_migration) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_auth_v2_migration)
> 
> #### Build Details:
> * Duration: 2 hr 28 min
> * Builder: spider2.cloudius-systems.com

#18118

**Comment by tchaikov** [comment]
@avikivity hi Avi, ping ?

**Comment by tchaikov** [comment]
@avikivity hi Avi, ping ?

**Comment by tchaikov** [comment]
@scylladb/scylla-maint @avikivity hello maintainers and Avi, could you help review this change?

File: docs/architecture/anti-entropy/read-repair.rst [hunk 1]
```diff
@@ -39,38 +39,8 @@ See the appendices below for the detailed flow.
...
-.. raw:: html
-
-	&nbsp;
-
-5. If a query’s consistency level is more than 1, and a data mismatch is found before consistency level is reached, then the read repair will run in the foreground. **All replica nodes are updated before the data is returned to the client.** However, if data mismatch  is found *after* a query’s consistency level is achieved, the update happens in the background, as mentioned above.
-
-The probability of this kind of read repair occurring on a read request is based on the :code:`read_repair_chance`. This :doc:`setting </cql/ddl/>`, provided when a table is created, can apply across data centers.  The setting specifies the probability that extra nodes (in addition to those required by the consistency level) are queried for the purpose of read repairs.
-
-A similar :doc:`setting </cql/ddl/>`, :code:`dclocal_read_repair_chance`,  applies only to nodes within the local data center.  You can use this setting when conserving traffic between data centers is important.
+Probabilistic read repair configured using ``read_repair_chance`` and ``dclocal_read_repair_chance`` in ``scylla.yaml`` is removed in Scylla 6.1.
```
Comment (denesb) [inline docs/architecture/anti-entropy/read-repair.rst:43]: ScyllaDB 6.0, no?

@annastuchlik I recall you have some guidelines on whether we should mention things like "removed in version X", as we now have per-version documentation anyway.

Comment (tchaikov) [reply docs/architecture/anti-entropy/read-repair.rst]: @denesb hi Botond, this magic (version) number came from #3502, where the milestone was marked 6.1. so i blindly use it. but i agree with you, before GA, the first release version in the 6.x branch should always be 6.0.

**Comment by tchaikov** [comment]
v3:

- s/6.1/6.0/. as 6.0 should be the next major release.

**Comment by tchaikov** [comment]
v4:

- drop the entire "Probabilistic Read Repair" section.

File: docs/architecture/anti-entropy/read-repair.rst (hunk 1 above)
Comment (tchaikov) [reply docs/architecture/anti-entropy/read-repair.rst]: @annastuchlik hi Anna, thank you for pointing this out. dropped.

**Review by denesb — APPROVED** [review]

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8297/) ✅ - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8297/) **The following new/updated tests ran 100 times for each mode:** 🔹 boost/config_test 🔹 boost/cql_query_test 🔹 boost/sstable_3_x_test 🔹 cql-pytest/test_describe ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8297/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/1205/) ❌ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/9227/) ✅ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/8297/)
> 
> #### Failed Tests (1/39835):
> * [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/8297/testReport/junit/rolling_upgrade_test/TestRollingUpgrade/Tests___dtest___test_rolling_upgrade) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
> 
> #### Build Details:
> * Duration: 8 hr 32 min
> * Builder: i-08eb7cef23dfc95f6 (m5ad.8xlarge)

#16002

File: docs/architecture/anti-entropy/read-repair.rst (hunk 1 above)
Comment (tchaikov) [reply docs/architecture/anti-entropy/read-repair.rst]: @dani-tweig hi Dani, how about adding a dedicated github label for the release-note worthy changes?
@tzach hi Tzach, we could have a `PendingReleaseNotes` file in the source repo, and require developer to document the change in this file. back when working on Ceph, we have https://github.com/ceph/ceph/blob/main/PendingReleaseNotes . it'd be more efficient to document these changes this way, what do you think?

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello maintainers, could you help queue this change?

**Pull request merged**